dash-renderer/src/actions/index.js

```diff
@@ -411,8 +442,45 @@ function fillVals(paths, layout, cb, specs, depType) {
             specs[i],
             cb.anyVals,
             depType
-        )
-    );
+        );
+
+    const tryFill = (inputList, i) => {
+        try {
+            const inputs = fillInputs(inputList, i);
+            if (isMultiValued(specs[i]) && !inputs.length) {
+                emptyMultiValues++;
+            }
+            return inputs;
+        } catch (e) {
+            if (e instanceof ReferenceError) {
+                errors.push(e);
+                return null;
+            }
```

Comment:
Wondering what would be required to be able to write this (and https://github.com/plotly/dash/pull/1212/files#diff-15692ba768eeab56e95aa409ab7c19b7R279) so that we don't use exceptions as flow control.

Reply:
> don't use exceptions as flow control

yeah I suppose that's cleaner now that they aren't always errors -> https://github.com/plotly/dash/pull/1212/commits/b44bd54a8cd6c2d3e0e89e73c2cbc190b499ec5a